spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkParquetWriter.java

```diff
@@ -50,7 +50,8 @@ public class TestSparkParquetWriter {
   private static final Schema SCHEMA =
       new Schema(
           Types.NestedField.required(1, "id", Types.IntegerType.get()),
-          Types.NestedField.required(2, "id_long", Types.LongType.get()));
+          Types.NestedField.required(2, "id_long", Types.LongType.get()),
+          Types.NestedField.required(3, "v", Types.VariantType.get()));
```

Comment:
This schema is only used for `testFpp`, and that doesn't even use the schema. It creates a writer and then validates the FPP setting using reflection. I think you can revert this since it doesn't actually test any variant code paths.

Reply:
Reverted. I was using more tests like testCorrectness() which will go through variant code. But it duplicates to `testTypeSchema()`.